ApacheDS: Cannot find the comparator class DeepTrimToLowerComparator

Getting following LdapSchemaException with ApacheDS. Update to API 2.0.0.AM2 seems to solve the problem.
```
org.apache.directory.api.ldap.model.exception.LdapUnwillingToPerformException: Cannot find the comparator class org.apache.directory.api.ldap.model.schema.comparators.DeepTrimToLowerComparator PluginClassLoader[org.majki.intellij.ldapbrowser, 1.4] com.intellij.ide.plugins.cl.PluginClassLoader@2421b3fe
	at org.apache.directory.api.ldap.schema.loader.SchemaEntityFactory.getLdapComparator(SchemaEntityFactory.java:649)
	at org.apache.directory.api.ldap.schema.manager.impl.DefaultSchemaManager.addComparators(DefaultSchemaManager.java:888)
	at org.apache.directory.api.ldap.schema.manager.impl.DefaultSchemaManager.addSchemaObjects(DefaultSchemaManager.java:299)
	at org.apache.directory.api.ldap.schema.manager.impl.DefaultSchemaManager.load(DefaultSchemaManager.java:819)
	at org.apache.directory.api.ldap.schema.manager.impl.DefaultSchemaManager.loadDepsFirst(DefaultSchemaManager.java:1312)
	at org.apache.directory.api.ldap.schema.manager.impl.DefaultSchemaManager.loadWithDeps(DefaultSchemaManager.java:1229)
	at org.apache.directory.api.ldap.schema.manager.impl.DefaultSchemaManager.loadAllEnabled(DefaultSchemaManager.java:1105)
	at org.apache.directory.ldap.client.api.LdapNetworkConnection.loadSchema(LdapNetworkConnection.java:3648)
	at org.apache.directory.ldap.client.api.LdapNetworkConnection.loadSchemaRelaxed(LdapNetworkConnection.java:3632)
	at org.majki.intellij.ldapbrowser.ldap.LdapObjectClass.getTop(LdapObjectClass.java:176)
	at org.majki.intellij.ldapbrowser.ldap.LdapNode.createRoot(LdapNode.java:338)
	at org.majki.intellij.ldapbrowser.ldap.ui.LdapServerTreeNode.createChildren(LdapServerTreeNode.java:29)
	at org.majki.intellij.ldapbrowser.ldap.ui.LdapServerTreeNode.getChildren(LdapServerTreeNode.java:38)
	at org.majki.intellij.ldapbrowser.ldap.ui.LdapServerTreeNode.getChildCount(LdapServerTreeNode.java:55)
	at javax.swing.tree.DefaultTreeModel.getChildCount(DefaultTreeModel.java:186)
	at javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1479)
	at javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1288)
	at javax.swing.tree.VariableHeightLayoutCache.ensurePathIsExpanded(VariableHeightLayoutCache.java:984)
	at javax.swing.tree.VariableHeightLayoutCache.setExpandedState(VariableHeightLayoutCache.java:182)
	at javax.swing.plaf.basic.BasicTreeUI.updateExpandedDescendants(BasicTreeUI.java:1696)
	at javax.swing.plaf.basic.BasicTreeUI$Handler.treeExpanded(BasicTreeUI.java:3805)
	at javax.swing.JTree.fireTreeExpanded(JTree.java:2771)
	at javax.swing.JTree.setExpandedState(JTree.java:3636)
	at javax.swing.JTree.expandPath(JTree.java:2219)
	at javax.swing.plaf.basic.BasicTreeUI.toggleExpandState(BasicTreeUI.java:2286)
	at javax.swing.plaf.basic.BasicTreeUI.handleExpandControlClick(BasicTreeUI.java:2273)
	at javax.swing.plaf.basic.BasicTreeUI.checkForClickInExpandControl(BasicTreeUI.java:2231)
	at javax.swing.plaf.basic.BasicTreeUI$Handler.handleSelection(BasicTreeUI.java:3600)
	at javax.swing.plaf.basic.BasicTreeUI$Handler.mousePressed(BasicTreeUI.java:3548)
	at com.intellij.util.ui.MouseEventAdapter.mousePressed(MouseEventAdapter.java:33)
	at java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:279)
	at java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:279)
	at java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:279)
	at java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:279)
	at java.awt.Component.processMouseEvent(Component.java:6547)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3325)
	at com.intellij.ui.treeStructure.Tree.processMouseEvent(Tree.java:398)
	at java.awt.Component.processEvent(Component.java:6315)
	at java.awt.Container.processEvent(Container.java:2239)
	at java.awt.Component.dispatchEventImpl(Component.java:4899)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Component.dispatchEvent(Component.java:4721)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4532)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
	at java.awt.Container.dispatchEventImpl(Container.java:2283)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4721)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:766)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
	at java.awt.EventQueue$4.run(EventQueue.java:739)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:736)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:741)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:686)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:385)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: org.apache.directory.api.ldap.model.exception.LdapSchemaException: Cannot find the comparator class org.apache.directory.api.ldap.model.schema.comparators.DeepTrimToLowerComparator PluginClassLoader[org.majki.intellij.ldapbrowser, 1.4] com.intellij.ide.plugins.cl.PluginClassLoader@2421b3fe
	at org.apache.directory.api.ldap.schema.loader.SchemaEntityFactory.classLoadComparator(SchemaEntityFactory.java:463)
	at org.apache.directory.api.ldap.schema.loader.SchemaEntityFactory.getLdapComparator(SchemaEntityFactory.java:639)
	... 68 more
```

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,6 +1,6 @@
 version=1.4
 ideaVersion=2018.1
 javaVersion=1.8
-apacheDirectoryApiVersion=1.0.2
+apacheDirectoryApiVersion=2.0.0.AM2
 
 org.gradle.logging.level=info

--- a/src/org/majki/intellij/ldapbrowser/ldap/LdapNode.java
+++ b/src/org/majki/intellij/ldapbrowser/ldap/LdapNode.java
@@ -102,8 +102,8 @@ public class LdapNode implements Serializable {
     }
 
     private void addValuesFromAttribute(Attribute attribute, List<LdapAttribute.Value> values) {
-        for (Value<?> value : attribute) {
-            values.add(new LdapAttribute.Value(value.isNull(), value.isHumanReadable(), value.getString(), value.getBytes()));
+        for (Value value : attribute) {
+            values.add(new LdapAttribute.Value(value.isNull(), value.isHumanReadable(), value.getValue(), value.getBytes()));
         }
     }
 
@@ -114,8 +114,8 @@ public class LdapNode implements Serializable {
         }
         for (Attribute attribute : rootDse.getAttributes()) {
             if (attribute.getId().equalsIgnoreCase(NAMING_CONTEXT_ATTRIBUTE_NAME)) {
-                for (Value<?> value : attribute) {
-                    LdapNode node = new LdapNode(ldapConnectionInfo, this, topObjectClass, value.getString(), value.getString(), new ArrayList<>());
+                for (Value value : attribute) {
+                    LdapNode node = new LdapNode(ldapConnectionInfo, this, topObjectClass, value.getValue(), value.getValue(), new ArrayList<>());
                     node.refresh();
                     children.add(node);
                 }

--- a/src/org/majki/intellij/ldapbrowser/ldap/LdapNode.java
+++ b/src/org/majki/intellij/ldapbrowser/ldap/LdapNode.java
@@ -129,15 +129,12 @@ public class LdapNode implements Serializable {
         if (dn == null || dn.trim().isEmpty()) {
             readRootDSN();
         } else {
-            try( EntryCursor cursor = getConnection().search(dn, DEFAULT_FILTER, SearchScope.ONELEVEL, "")) {
-                for (int i=0;cursor.next();i++) {
+            try( EntryCursor cursor = getConnection().search(dn, DEFAULT_FILTER, SearchScope.ONELEVEL, DEFAULT_SEARCH_ATTRIBUTE)) {
+                while (cursor.next()) {
                     Entry entry = cursor.get();
                     List<LdapAttribute> attributes = new ArrayList<>();
-                    if(i<100) {
-                        Entry attributeHolder = getConnection().lookup(entry.getDn());
-                        for (Attribute attribute : attributeHolder.getAttributes()) {
-                            mapLdapAttributes(attributes, attribute);
-                        }
+                    for (Attribute attribute : entry.getAttributes()) {
+                        mapLdapAttributes(attributes, attribute);
                     }
                     children.add(new LdapNode(ldapConnectionInfo, this, topObjectClass, entry.getDn().getName(), entry.getDn().getRdn().getName(), attributes));
                 }

